src/appshell/view/appmenumodel.cpp

```diff
@@ -199,35 +239,41 @@ MenuItem AppMenuModel::editItem()
 MenuItem AppMenuModel::viewItem()
 {
     MenuItemList viewItems {
-        makeAction("toggle-palette", canUndo()), // need implement
-        makeAction("masterpalette", canRedo()), // need implement
-        makeAction("inspector", selectedElementOnScore()), // need implement
-        makeAction("toggle-playpanel", selectedElementOnScore()), // need implement
-        makeAction("toggle-navigator", selectedElementOnScore()), // need implement
-        makeAction("toggle-timeline", selectedElementOnScore()), // need implement
-        makeAction("toggle-mixer", selectedElementOnScore()), // need implement
-        makeAction("synth-control", selectedElementOnScore()), // need implement
-        makeAction("toggle-selection-window", selectedElementOnScore()), // need implement
-        makeAction("toggle-piano", selectedElementOnScore()), // need implement
-        makeAction("toggle-scorecmp-tool", scoreOpened()), // need implement
+        makeAction("toggle-palette", isNotationPage(), configuration()->isPalettePanelVisible().val),
```

Comment:
I believe we should do all those checks in concrete modules. For example, palettes module register "toggle-palette" action in PaletteActionsController. Then PaletteActionsController should answer that it is possible to use this action now. We already have an interface for this - Actionable::canReceiveAction

@igorkorsukov what do you think?

Reply:
I will fix it in next PR